scripts/perturbation_driver.py

```diff
@@ -0,0 +1,299 @@
+        toc = time.perf_counter()
+        elapsed_time = toc - tic
+        print(f"Elapsed time: {elapsed_time:0.4f} seconds")
+        return value
+    return wrapper_timer
+
+@timer
+def main(cla):
+
+    ''' Main function that takes a Namespace structure of command line
+    arguments (cla) and generates the requested ensemble output. '''
+
+    for fhr in cla.fcst_hour:
+
+        # Call the appropriate variables function
+        variables = globals().get(f"{cla.vars}_variables")()
+
+        # Get perturbations
+        if cla.inputpath:
+            ens_path = cla.inputpath.format(fhr=fhr)
+            ens_perts = compute_perturbations(ens_path, variables)
+            ens_mean = compute_ens_mean(ens_path, variables)
+
+        elif cla.perturbation_file:
+            ens_perts = load_perturbations(cla.perturbation_file)
+
+        elif cla.ens_mean_file:
+            ens_mean = load_perturbations(cla.ens_mean_file)
+
+        else:
+            print('No source of ensemble perturbations was provided. ' +
+                  'Please call script with -p or -i option')
+
+        # Write ens perturbations to a single file
+        if cla.write_perturbations:
+            if cla.perturbation_file:
+                ens_perts.to_netcdf(cla.perturbation_file)
+
+        # Write ens mean to a single file
+        if cla.write_ens_mean:
+            if cla.ens_mean_file:
+                ens_mean.to_netcdf(cla.ens_mean_file)
+
+        # Write ensmeble of full-state files
+        if cla.ens_outfn_tmpl and cla.base_state:
+
+            # Open base state file
+            base_fpath = cla.base_state.format(fhr=fhr)
+
+            for mem in range(0, ens_perts.dims['ens']):
+
+                base_state = xr.open_mfdataset(base_fpath)
+                print(f'Preparing member {mem+1}')
+                # Create output directory. Done here to allow support
+                # for mem field in format string.
+                outputdir = cla.outputdir.format(mem=mem+1)
+                os.makedirs(outputdir, exist_ok=True)
+
+                # Identify member output file, and generate a list
+                mem_fname = cla.ens_outfn_tmpl.format(fhr=fhr, mem=mem+1)
+                mem_fpath = os.path.join(outputdir, mem_fname)
+
+                # Write full ensemble members to disk. One at a time allows
+                # compression, and should give similar performance as
+                # save_mfdataset() without a dask cluster.
+                base_state.update(base_state +
+                        ens_perts[variables].sel(ens=mem))
+                comp = {'zlib': True, 'complevel': 9}
+                encoding = {var: comp for var in base_state.data_vars}
+                base_state.to_netcdf(mem_fpath, encoding=encoding)
+                base_state.close()
+
+def atmo_variables():
+
+    ''' Return list of initial atmospheric variables to be perturbed. '''
+
+    return ['ps', 't', 'zh', 'sphum', 'u_w', 'v_w', 'u_s', 'v_s']
```

Comment:
It might be easier for future code management activities to keep these lists in alphabetical order, and one item per line. This script was a prototype in its original form, and wasn't meant to go "operational" without some additional attention. ;)

Reply:
Yes, it does look better in alphabetical order. Will update them accordingly. This script looks pretty good to me but some additional developments would be greatly appreciated! Thank you very much!